File: python/matrix/matrix.py
class Matrix(object):
    def __init__(self, matrix_string):
        self.mat = [
            [int(i) for i in row.split()] for row in matrix_string.split("\n")
        ]  # this is the matrix as list of lists of elements

    def row(self, index):
        return self.mat[index - 1]

    def column(self, index):
        return [row[index - 1] for row in self.mat]

File: javascript/matrix/matrix.js
export class Matrix {
    constructor(matrixString) {

        this.row = matrixString.split('\n').map(x => x.split(' ').map(x => Number(x)));
    }

    get rows() {
        return this.row;
    }

    get columns() {

        let cols = [];
        for (let i = 0; i < this.row[0].length; i++) {
            cols.push(this.row.map(x => x[i]));
        }
        return cols;

    }
}
